app/memorial/controllers/edit.py

```diff
@@ -1,3 +1,41 @@
+from app.memorial.repos import MemorialRepo, RoleRepo
+from app.services.permission import Action, can_user_execute
 
-def edit_memorial(creator_id, description, first_name, last_name):
-    pass
\ No newline at end of file
+
+def edit_memorial(user_id, memorial_id, first_name, last_name, description, born, died, bio, home_town, cover_photo,
+                  page_theme, can_post, can_view, can_manage, can_delete):
+    # 1. Check valid user
+    # 2. Display all user memorials based on user_id
+    # 3. Actual memorial_id
+    # 4. Change memorial based on user_id and memorial_id
+    # 5. Change the content based on memorial
+
+    if not user_id:
+        return {"msg": "Missing user"}, 400
+
+    memorial = MemorialRepo.get_by_id(memorial_id)
+
+    if not can_user_execute(Action.EDIT_PRIVACY, RoleRepo.get_by_user_id(user_id), memorial):
+        return {"msg": "Invalid permission"}, 400
```

Comment:
Whoops, I forgot to delete the `EDIT_PRIVACY` action. We should use `MANAGE` here. Could you also delete EDIT_PRIVACY in the action enum in `app/services/permission.py`?

This will crash since there's no `canEditPrivacy` property on the memorial anymore.

Reply:
done